### PR TITLE
Web

### PR DESCRIPTION
**Pull request by malcheg**

### CODE REVIEW

File: frontend/src/main/scala/io/udash/homepage/components/Buttons.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import io.udash.homepage.styles.partials.ButtonsStyle
+
+import scalatags.JsDom.all._
+import scalacss.ScalatagsCss._
+
+/**
+  * Created by malchik on 2016-04-04.
+  */
+object Buttons {
+  def whiteBorderButton(link: String, label: String, xs: Modifier*) = a(href := link, target := "_blank", ButtonsStyle.btnDefault, xs: Modifier)(
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Buttons.scala]: Move `a(href := link, target := "_blank", ButtonsStyle.btnDefault, xs: Modifier)(` to the new line.

File: frontend/src/main/scala/io/udash/homepage/components/CodeBlock.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import io.udash.homepage.styles.partials.HomepageStyles
+import org.scalajs.dom
+import org.scalajs.dom.raw.Element
+
+import scalatags.JsDom
+import scalatags.JsDom.all._
+import scalacss.ScalatagsCss._
+
+object CodeBlock {
+  def apply(data: String, xs: Modifier*): Element =
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/CodeBlock.scala]: IMO `"language-scala"` passed as `cls` to `code` should be `def apply` parameter with default value.

```scala
def apply(data: String, language: String = "language-scala")(xs: Modifier*)
```

File: frontend/src/main/scala/io/udash/homepage/components/Footer.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.scalajs.dom.raw.Element
+
+import scalatags.JsDom.all._
+import scalacss.ScalatagsCss._
+
+class Footer() {
+  def getTemplate: Element = {
+    footer(FooterStyles.footer)(
+      div(GlobalStyles.body, GlobalStyles.clearfix)(
+        a(FooterStyles.footerLogo, href := "/")(
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Footer.scala]:
```scala
href := IndexState.url
```

File: frontend/src/main/scala/io/udash/homepage/components/Footer.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          img(GlobalStyles.block, src := "assets/images/udash_logo.png", alt := "Udash Guide")
+        ),
+        div(FooterStyles.footerLinks)(
+          p(FooterStyles.footerMore)("See more"),
+          ul(
+            li(FooterStyles.footerLinkWrapper)(
+              a(href := "https://github.com/UdashFramework/udash-demos", target := "_blank", FooterStyles.footerLink)("Github demo")
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Footer.scala]: Create object with these links, please.

```scala
object ExternalUrls {
  val udashDemos = "https://github.com/UdashFramework/udash-demos"
  val stackoverflow = "http://stackoverflow.com/questions/tagged/udash"
  val avsystem = "http://www.avsystem.com/"
  // etc...
}
```

File: frontend/src/main/scala/io/udash/homepage/components/Footer.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            )
+          ),
+          p(FooterStyles.footerCopyrights)("Udash was made with great support of ", a(FooterStyles.footerAvsystem, href := "http://www.avsystem.com/")("AVSystem"))
+        )
+      )
+    ).render
+  }
+}
+
+object Footer {
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Footer.scala]: Why you do not simply change Footer class to object?

File: frontend/src/main/scala/io/udash/homepage/components/Header.scala [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            li(HeaderStyles.headerLinkWrapper)(
+              a(href := "/", HeaderStyles.headerLink)("Documentation")
+            )
+          )
+        )*/
+      ),
+      div(HeaderStyles.headerRight)(
+        ul(HeaderStyles.headerSocial)(
+          li(HeaderStyles.headerSocialItem)(
+            a(href := "https://github.com/UdashFramework", HeaderStyles.headerSocialLink, target := "_blank")(
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Header.scala]: Here you can use `ExternalUrls` object which I mentioned before.

File: frontend/src/main/scala/io/udash/homepage/components/demo/DemoComponent.scala [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+class DemoComponent(url: Property[String]) extends Component {
+
+  url.listen(onUrlChange)
+  Window.onResize(onResize)
+
+  private def onUrlChange(update: String) = {
+    val entryOption = DemoComponent.demoEntries.find(_.url.substring(1) == update)
+    val entry = entryOption.getOrElse(DemoComponent.demoEntries.head)
+    val urlString = "\"" + entry.url + "\""
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/demo/DemoComponent.scala]:
```scala
val urlString = s""""${entry.url}""""
```

File: frontend/src/main/scala/io/udash/homepage/components/demo/DemoPreview.scala [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+  def validationDemo: Element = {
+    import io.udash._
+
+    import scalajs.concurrent.JSExecutionContext.Implicits
+    import Implicits.queue
+    import scalacss.ScalatagsCss._
+    import scalatags.JsDom.all._
+
+    val emailRegex = "([\\w.%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,})".r
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/demo/DemoPreview.scala]:
```scala
val emailRegex = "([\\w.%+-]+@[\\w.-]+\\.[a-zA-Z]{2,})".r
```

File: frontend/src/main/scala/io/udash/homepage/states.scala [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+sealed abstract class RoutingState(val parentState: RoutingState) extends State {
+  def url(implicit application: Application[RoutingState]): String = s"#${application.matchState(this).value}"
+}
+
+case object RootState extends RoutingState(null)
+
+case object ErrorState extends RoutingState(RootState)
+
+case class IndexState(additionalArgument: Option[String]) extends RoutingState(RootState)
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/states.scala]: `additionalArgument` -> `selectedDemoTab` or something like that.

File: frontend/src/main/scala/io/udash/homepage/styles/GlobalStyles.scala [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+
+    unsafeRoot("svg") (
+      display.block
+    ),
+
+    unsafeRoot("object[type='image/svg+xml']") (
+      pointerEvents := "none"
+    ),
+
+    unsafeRoot("object[type=\"image/svg+xml\"]") (
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/styles/GlobalStyles.scala]: `unsafeRoot("object[type='image/svg+xml']")` and `unsafeRoot("object[type=\"image/svg+xml\"]")` are not the same?

File: frontend/src/main/scala/io/udash/homepage/components/Footer.scala [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+import scalatags.JsDom.all._
+import scalacss.ScalatagsCss._
+
+class Footer() {
+  def getTemplate: Element = {
+    footer(FooterStyles.footer)(
+      div(GlobalStyles.body, GlobalStyles.clearfix)(
+        a(FooterStyles.footerLogo, href := "/")(
+          img(GlobalStyles.block, src := "assets/images/udash_logo.png", alt := "Udash Guide")
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/components/Footer.scala]: I think that you can use `Image` object here.

File: frontend/src/main/scala/io/udash/homepage/wrappers/scrollbar/package.scala [hunk 12]
```diff
@@ -0,0 +1,26 @@
+package io.udash.homepage.wrappers
```
Comment (Starzu) [inline frontend/src/main/scala/io/udash/homepage/wrappers/scrollbar/package.scala]: Move this wrapper to `io.udash.wrappers.jquery.scrollbar` package.

**Comment by malcheg** [comment]
take a look, please

**Pull request merged**